courses-app/src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -1,33 +1,41 @@
-import React from 'react';
-import { useRecoilState, useResetRecoilState } from 'recoil';
+import { React, useState, useEffect } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { v4 as uuidv4 } from 'uuid';
 import Button from '../common/Button/Button';
 import Input from '../common/Input/input';
 import '../../App.css';
 import ListedAuthors from './components/ListedAuthors/ListedAuthors';
 import AddAuthor from './components/AddAuthor/AddAuthor';
 import AddDuration from './components/AddDuration/AddDuration';
 import CourseAuthors from './components/CourseAuthors/CourseAuthors';
-import newCourse from '../../recoil/atom/newCourse';
-import courses from '../../recoil/atom/courses';
+import { getAuthors, getCourses, createCourse } from '../../services/CoursesService';
 // import { mockedAuthorsList, mockedCoursesList } from '../../mockedCoursesList';
 
 function CreateCourse() {
   const navigate = useNavigate();
-  const reset = useResetRecoilState(newCourse);
+  const [authorsState, setAuthorsState] = useState([]);
   // eslint-disable-next-line no-unused-vars
-  const [newCourseState, setNewCourseState] = useRecoilState(newCourse);
-  const [coursesState, setCoursesState] = useRecoilState(courses);
+  const [newCourseState, setNewCourseState] = useState({
+    title: '', description: '', authors: [], duration: 0,
+  });
+  const [coursesState, setCoursesState] = useState([]);
+  useEffect(() => {
+    getAuthors().then((data) => {
```

Comment:
koristi async await umesto callback-ova (then catch). kreiraj posebne funkcije kako to ne bi pisala unutra useEffecta.